thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/DetectorHttpUtils.java

```diff
@@ -18,8 +20,12 @@
   private static String EMAIL_JOBS_ENDPOINT = "/api/email-jobs/";
   private static String ADHOC = "/ad-hoc";
 
-  public DetectorHttpUtils(String detectorHost, int detectorPort) {
+  public DetectorHttpUtils(String detectorHost, int detectorPort, String authToken) {
     super(new HttpHost(detectorHost, detectorPort));
+    BasicClientCookie cookie = new BasicClientCookie("te_auth", authToken);
```

Comment:
why is this code duplicated four times?

Reply:
It's for generating the cookie with authentication tokens. I do this way so that user only need to worry about the authentication token, and the constructor makes the cookie and insert into the request. I leave more flexibility for adding more cookie in the future.

Reply:
Thanks for the comment. The duplicate lines are removed.